src/main/java/qupath/AnnotationExchangeExtension/ImportAnnotationServiceJSONPlugin.java

```diff
@@ -88,111 +89,104 @@ private boolean readJSONAnnotations(File inputFile, ImageData imageData ) {
             JsonReader jsonReader = new JsonReader(new FileReader(inputFile));
             JsonParser jsonParser = new JsonParser();
 
-            JsonArray jsonAnnotationData = jsonParser.parse(jsonReader)
-                    .getAsJsonArray();
+            JsonObject jsonAnnotationData = jsonParser.parse(jsonReader).getAsJsonObject();
+
+            JsonArray jsonAnnotations = jsonAnnotationData.get("dictionaries").getAsJsonArray();
+
+            //Loop through every annotation in the dictionary
+            for (JsonElement jsonAnnotation : jsonAnnotations) {
+                //At the moment we were assuming that the name corresponds to tissue type so we will try to match it up with QuPath types and if none match, create a new one
+                String annotationName = jsonAnnotation.getAsJsonObject().get("name").getAsString();
+
+                // Dictionaries are created in the Annotation Service JS library. There isn't a clear convention on what they should mean but we will
+                // add a parameter to their imported objects so that they can be used later
+                String annotationDictionary = jsonAnnotation.getAsJsonObject().get("dictionary").getAsString();
+
+                // Dictionaries are created in the Annotation Service JS library. There isn't a clear convention on what they should mean but we will
+                // add a parameter to their imported objects so that they can be used later
+                String annotationUID = jsonAnnotation.getAsJsonObject().get("uid").getAsString();
 
-            JsonArray jsonAnnotationDictionaries = jsonAnnotationData.get(1)
-                    .getAsJsonObject().get("dictionaries")
+                JsonArray annotationColor = jsonAnnotation.getAsJsonObject().get("path")
+                    .getAsJsonObject().get("fillColor")
                     .getAsJsonArray();
 
-            //Loop though all the "dictionaries"
-            for (JsonElement jsonAnnotationDictionary : jsonAnnotationDictionaries) {
```

Comment:
The `diff` here looks really big, but really I just removed this outer `for` loop as a result of the `dictionaries` key getting streamlined from:

```jsonc
{
  // ...
  "dictionaries": [
    [
      {
         // ...
      }
      // ...
    ]
  ]
}
```

to

```jsonc
{
  // ...
  "dictionaries": [
      {
         // ...
      }
      // ...
  ]
}
```